File: src/main/java/com/four9ebays/service/ReturnRequestService.java
package com.four9ebays.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.four9ebays.domain.ReturnRequest;
import com.four9ebays.dto.ReturnRequestDTO;
import com.four9ebays.dto.ReturnRequestSearchDTO;
import com.four9ebays.dto.ReturnRequestPageDTO;
import com.four9ebays.dto.ReturnRequestConvertCriteriaDTO;
import com.four9ebays.service.GenericService;
import com.four9ebays.dto.common.RequestDTO;
import com.four9ebays.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ReturnRequestService extends GenericService<ReturnRequest, Integer> {

	List<ReturnRequest> findAll();

	ResultDTO addReturnRequest(ReturnRequestDTO returnRequestDTO, RequestDTO requestDTO);

	ResultDTO updateReturnRequest(ReturnRequestDTO returnRequestDTO, RequestDTO requestDTO);

    Page<ReturnRequest> getAllReturnRequests(Pageable pageable);

    Page<ReturnRequest> getAllReturnRequests(Specification<ReturnRequest> spec, Pageable pageable);

	ResponseEntity<ReturnRequestPageDTO> getReturnRequests(ReturnRequestSearchDTO returnRequestSearchDTO);
	
	List<ReturnRequestDTO> convertReturnRequestsToReturnRequestDTOs(List<ReturnRequest> returnRequests, ReturnRequestConvertCriteriaDTO convertCriteria);

	ReturnRequestDTO getReturnRequestDTOById(Integer returnRequestId);







}

File: four9ebays_ui/src/services/ReturnRequestService.js
import http from "../http-common"; 

class ReturnRequestService {
  getAllReturnRequests(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/returnRequest/returnRequests`, searchDTO);
  }

  get(returnRequestId) {
    return this.getRequest(`/returnRequest/${returnRequestId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/returnRequest?field=${matchData}`, null);
  }

  addReturnRequest(data) {
    return http.post("/returnRequest/addReturnRequest", data);
  }

  update(data) {
  	return http.post("/returnRequest/updateReturnRequest", data);
  }
  
  uploadImage(data,returnRequestId) {
  	return http.postForm("/returnRequest/uploadImage/"+returnRequestId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ReturnRequestService();
